static/app/views/ddm/useGetChartPalette.tsx

```diff
@@ -0,0 +1,62 @@
+import {useCallback, useRef} from 'react';
+
+import theme from 'sentry/utils/theme';
+
+const CACHE_SIZE = 20; // number of palettes to cache
+
+// We do length - 2 to be aligned with the colors in other parts of the app (copy-pasta)
+// We use Math.max to avoid numbers < -1 as then `getColorPalette` returns undefined (not typesafe because of array access)
+export function getChartColors(seriesCount: number) {
+  return theme.charts.getColorPalette(Math.max(seriesCount - 2, -1));
+}
+
+export function getCachedChartPalette(
+  cache: Record<string, string>[],
+  seriesNames: string[]
+): (seriesName: string) => string {
+  // We do length - 2 to be aligned with the colors in other parts of the app (copy-pasta)
+  // We use Math.max to avoid numbers < -1 as then `getColorPalette` returns undefined (not typesafe because of array access)
+  const defaultColors = getChartColors(seriesNames.length);
+
+  // Check if we already have a palette that includes all of the given seriesNames
+  // We reverse the cache to get the most recent palettes first
+  const cachedPalette = [...cache]
```

Comment:
I dont think you need shallow cloning here since you are just running `.find` which does not mutate the array,

Reply:
There is `reverse` which mutates the array. Can get rid of that if I use a for loop for iterating it in reverse, which would also  be a bit faster.